light_prop/utils.py

```diff
@@ -0,0 +1,48 @@
+import logging
+import numpy as np
+from PIL import Image
+
+
+def image_to_amplitude(image_path):
+    array = ndarray_to_image(image_path)
+    return normalize_array(array, 1)
+
+
+def image_to_phase(image_path):
+    array = ndarray_to_image(image_path)
+    return normalize_array(array, 2 * np.pi)
+
+
+def amplitude_to_image(array: np.array, file_path="output", file_extension="bmp"):
+    array = normalize_array(array, 255)
+    ndarray_to_image(array, file_path, file_extension)
+
+
+def phase_to_image(array: np.array, file_path="output", file_extension="bmp"):
+    array = array % 2 * np.pi
+    array = normalize_array(array, 255)
+    ndarray_to_image(array, file_path, file_extension)
+
+
+def ndarray_to_image(array: np.array, file_path="output", file_extension="bmp"):
+    path = f"{file_path}.{file_extension}"
+    logging.info("Reading array from ")
+    im = Image.fromarray(array)
+
+    if im.mode != 'RGB':
+        im = im.convert('RGB')
+    im.save(path)
+
+
+def image_to_ndarray(image_path):
+    array = np.asarray(Image.open(image_path))
+    # convert rgb 3dimensional matrix to 1d bw
+    return np.add.reduce(array, axis=2) / 3
```

Comment:
This won't always work I believe. I think the information about the datatype from image must be taken into account. Eg.: what if image is RGBA/ARGB?

Reply:
Another test to cover, gj!